genienlp/data_utils/remote_database.py

```diff
@@ -0,0 +1,279 @@
+import ujson
+import logging
+import nltk
+import time
+from collections import defaultdict
+
+from elasticsearch.client import XPackClient
+from elasticsearch.client.utils import NamespacedClient
+from elasticsearch import Elasticsearch, RequestsHttpConnection
+from elasticsearch import exceptions
+from .database import is_banned
+from .database_utils import normalize_text, has_overlap
+
+ES_RETRY_ATTEMPTS = 5
+
+tracer = logging.getLogger('elasticsearch')
+tracer.setLevel(logging.CRITICAL)
+
+logger = logging.getLogger(__name__)
+
+class XPackClientMPCompatible(XPackClient):
+    def __getattr__(self, attr_name):
+        return NamespacedClient.__getattribute__(self, attr_name)
+
+class RemoteElasticDatabase(object):
+    
+    def __init__(self, config, type2id, features_default_val, features_size):
```

Comment:
How is the config passed? Is it environment variables or arguments? Given it contains secrets, it should be either a config file or environment variables.

Reply:
It's a config file and should be located in `database_dir/es_material/elastic_config.json`